pkgs/dart_services/lib/src/sdk.dart

```diff
@@ -43,45 +36,62 @@ class Sdk {
   bool get mainChannel => _channel == 'main';
 
   // Which channel is this SDK?
-  final String _channel;
+  String? _channel;
+
+  // The directory that contains this SDK
+  String? _sdkPath;
 
   /// Experiments that this SDK is configured with
   List<String> get experiments {
     if (mainChannel) return const ['inline-class'];
     return const [];
   }
 
-  factory Sdk.create(String channel) {
-    final sdkPath = path.join(Sdk._flutterSdksPath, channel);
+  // When running with `dart run`, the FLUTTER_ROOT environment variable is
+  // set automatically.
+  String _getSdkPath() {
+    final env = Platform.environment;
+    if (!env.containsKey('FLUTTER_ROOT') || env['FLUTTER_ROOT']!.isEmpty) {
+      throw Exception('No FLUTTER_ROOT variable set');
+    }
+
+    return env['FLUTTER_ROOT']!;
+  }
+
+  String get sdkPath {
+    _sdkPath ??= _getSdkPath();
```

Comment:
Would this work as well?

```
String get sdkPath => (_sdkPath ??= _getSdkPath());
```

Reply:
I think `String get sdkPath => _sdkPath ??= _getSdkPath();` works.